newIDE/app/src/EventsBasedBehaviorEditor/EventsBasedBehaviorPropertiesEditor.js

```diff
@@ -267,6 +272,35 @@ export default class EventsBasedBehaviorPropertiesEditor extends React.Component
                                 />
                               </SelectField>
                             )}
+                            {property.getType() === 'Behavior' && (
+                              <BehaviorTypeSelector
+                                project={this.props.project}
+                                objectType={this.props.eventsBasedBehavior.getObjectType()}
+                                value={
+                                  property.getExtraInfo().size() === 0
+                                    ? ''
+                                    : property.getExtraInfo().at(0)
+                                }
+                                onChange={(newValue: string) => {
+                                  if (property.getExtraInfo().size() === 0) {
+                                    property.addExtraInfo(newValue);
+                                  } else {
+                                    //TODO This doesn't work because extraInfo is constant.
+                                    // Maybe this is a hack and the behavior type must be set somewhere else?
+                                    // Should a method be added to replace a value?
+                                    console.log('newValue: ' + newValue);
+                                    property.getExtraInfo().clear();
+                                    property.addExtraInfo(newValue);
+                                    console.log(
+                                      'get: ' + property.getExtraInfo().at(0)
+                                    );
```

Comment:
I used the property extra informations to store the required behavior type, but maybe that not a good idea because it seems that values can only add up.

Reply:
I added a `clearExtraInfo` method but that may not be the good call.

Reply:
I think it is fine to add a way to modify the extra data, you could also add a `void setExtraData(size_t index, const gd::String& extraData);` method to have a cleaner way of doing what you do.

Reply:
I was wondering if this should just not be constant instead of redirecting the whole vector interface.